hangman.py

```diff
@@ -0,0 +1,340 @@
+
+
+
+def load_words():
+    """
+    Returns a list of valid words. Words are strings of lowercase letters.
+    
+    Depending on the size of the word list, this function may
+    take a while to finish.
+    """
+    print("Loading word list from file...")
+    # inFile: file
+    inFile = open(WORDLIST_FILENAME, 'r')
+    # line: string
+    line = inFile.readline()
+    # wordlist: list of strings
+    wordlist = line.split()
+    print("  ", len(wordlist), "words loaded.")
+    return wordlist
+
+
+
+def choose_word(wordlist):
+    """
+    wordlist (list): list of words (strings)
+    
+    Returns a word from wordlist at random
+    """
+    return random.choice(wordlist)
+
+
+wordlist = load_words()
+
+
+def is_word_guessed(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing; assumes all letters are
+      lowercase
+    letters_guessed: list (of letters), which letters have been guessed so far;
+      assumes that all letters are lowercase
+    returns: boolean, True if all the letters of secret_word are in letters_guessed;
+      False otherwise
+    '''
+    letters_guessed = set(letters_guessed)
+    new_set = set(secret_word)
+    if letters_guessed | new_set == letters_guessed:
+    #This command combine two sets, if  whole new_set is in letters_guessed, this will give us set letters_guessed
+    #If whole new_set is not in letters_guessed, than it will give us set, which is bigger, than letters_guessed
+        return True
+    else:
+        return False
+
+
+
+def get_guessed_word(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string, comprised of letters, underscores (_), and spaces that represents
+      which letters in secret_word have been guessed so far.
+    '''
+
+    new_list = []
+    for symbol in secret_word:
+        if symbol in letters_guessed:
+            new_list.append(symbol)
+        else:
+            new_list.append('_ ')
+    if new_list[-1] == '_ ':
+        new_list[-1] = '_'
+    str = ''.join(new_list)
+    return str
+
+
+
+def get_available_letters(letters_guessed):
+    '''
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string (of letters), comprised of letters that represents which letters have not
+      yet been guessed.
+    '''
+
+    end_list = []
+    for letters in list(string.ascii_lowercase):
+        if letters in letters_guessed:
+            continue
+        else:
+            end_list.append(letters)
+
+    str_not_used_letters = ''.join(end_list)
+    return str_not_used_letters
+
+
+def len_secret_word_without_dublicate(secret_word):
+    new_set = set()
+    for symbol in secret_word:
+        new_set.add(symbol)
+    length = len(new_set)
+    return length
+    
+    
+
+def hangman(secret_word, hints_switch = False):
```

Comment:
This function is still quite long. Try extract some methods here. Try to achieve up to 30 lines (including empty lines and comments)